examples/platform/qpg6100/app/Service.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ *   This file implements the service publishing code for example usage.
+ */
+
+#include "Service.h"
+
+#include "AppConfig.h"
+#include "FreeRTOS.h"
+#include "task.h"
+#include <string.h>
+#include <sys/param.h>
+
+#include "lwip/err.h"
+#include "lwip/sockets.h"
+#include "lwip/sys.h"
+#include <lwip/netdb.h>
+
+#include <inet/IPAddress.h>
+#include <inet/InetError.h>
+#include <inet/InetLayer.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <support/CodeUtils.h>
+#include <support/ErrorStr.h>
+#include <system/SystemPacketBuffer.h>
+#include <transport/SecureSessionMgr.h>
+#include <transport/raw/UDP.h>
+
+#if CHIP_ENABLE_OPENTHREAD
+#include <openthread/message.h>
+#include <openthread/udp.h>
+#include <platform/OpenThread/OpenThreadUtils.h>
+#include <platform/ThreadStackManager.h>
+#include <platform/internal/DeviceNetworkInfo.h>
+#include <platform/qpg6100/ThreadStackManagerImpl.h>
+#endif
+
+using namespace ::chip;
+using namespace ::chip::Inet;
+using namespace ::chip::Transport;
+using namespace ::chip::DeviceLayer;
+
+static char sDeviceName[128] = { 0 };
+// Hardcoded UDP BroadcastPort. Temporary use for demo with OTBR
+constexpr uint16_t kUDPBroadcastPort = 23367;
+
+void SetDeviceName(const char * newDeviceName)
+{
+    strncpy(sDeviceName, newDeviceName, sizeof(sDeviceName) - 1);
+}
+
+void PublishService()
+{
+    chip::Inet::IPAddress addr;
+    if (!ConnectivityMgrImpl().IsThreadAttached())
+    {
+        return;
+    }
+    ThreadStackMgrImpl().LockThreadStack();
+    otError error = OT_ERROR_NONE;
+    otMessageInfo messageInfo;
+    otUdpSocket mSocket;
```

Comment:
It's a bit weird to name this `mSocket` if it's not a member...

Reply:
correct - have to admit this is inherited code - that could better be moved to a common location